v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from dataclasses import dataclass
+
+import hail as hl
+
+from v03_pipeline.lib.model import ReferenceGenome
+from v03_pipeline.lib.reference_data.config import CONFIG
+from v03_pipeline.lib.reference_data.dataset_table_operations import (
+    get_all_select_fields,
+    get_ht_path,
+    import_ht_from_config_path,
+    parse_dataset_version,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ReferenceDataGlobals:
+    paths: dict[str]
+    versions: dict[str]
+    enums: dict[str, dict[str, list[str]]]
+
+    def __init__(self, globals_struct: hl.Struct):
+        self.paths = self._struct_to_dict(globals_struct.paths)
+        self.versions = self._struct_to_dict(globals_struct.versions)
+        self.enums = self._struct_to_dict(globals_struct.enums)
+
+    def _struct_to_dict(self, struct: hl.Struct) -> dict:
+        result_dict = {}
+        for field in struct:
+            if isinstance(struct[field], hl.Struct):
+                result_dict[field] = self._struct_to_dict(struct[field])
+            else:
+                result_dict[field] = struct[field]
+        return result_dict
+
+
+def get_datasets_to_update(
+    joined_ht: hl.Table,
+    datasets: list[str],
+    reference_genome: ReferenceGenome,
+) -> list[str]:
+    datasets_to_update = []
+    for dataset in datasets:
+        if dataset not in joined_ht.row:
+            datasets_to_update.append(dataset)
+            continue
+
+        if not validate_joined_ht_globals_match_config(
+            joined_ht,
+            dataset,
+            reference_genome,
+        ):
+            datasets_to_update.append(dataset)
+    return datasets_to_update
+
+
+def validate_joined_ht_globals_match_config(
+    joined_ht: hl.Table,
+    dataset: str,
+    reference_genome: ReferenceGenome,
+) -> bool:
+    joined_ht_globals = ReferenceDataGlobals(hl.eval(joined_ht.index_globals()))
```

Comment:
Can we lift this eval up one more function?  I think we want to call the eval of the globals just one single time, not once per dataset!

Reply:
so true, will do